openfeature/multiprovider/multiprovider.go

```diff
@@ -0,0 +1,569 @@
+// Package multiprovider implements an OpenFeature provider that supports multiple feature flag providers.
+package multiprovider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"slices"
+	"strings"
+	"sync"
+	"time"
+
+	of "github.com/open-feature/go-sdk/openfeature"
+	"golang.org/x/sync/errgroup"
+)
+
+// Metadata Keys
+const (
+	MetadataProviderName                   = "multiprovider-provider-name"
+	MetadataProviderType                   = "multiprovider-provider-type"
+	MetadataInternalError                  = "multiprovider-internal-error"
+	MetadataSuccessfulProviderName         = "multiprovider-successful-provider-name"
+	MetadataSuccessfulProviderNames        = MetadataSuccessfulProviderName + "s"
+	MetadataStrategyUsed                   = "multiprovider-strategy-used"
+	MetadataFallbackUsed                   = "multiprovider-fallback-used"
+	MetadataIsDefaultValue                 = "multiprovider-is-result-default-value"
+	MetadataEvaluationError                = "multiprovider-evaluation-error"
+	MetadataComparisonDisagreeingProviders = "multiprovider-comparison-disagreeing-providers"
+)
+
+type (
+	// ProviderMap is an alias for a map containing unique names for each included [of.FeatureProvider]
+	ProviderMap = map[string]of.FeatureProvider
+
+	// MultiProvider is an implementation of [of.FeatureProvider] that can execute multiple providers using various
+	// strategies.
+	MultiProvider struct {
```

Comment:
Because the name of the package is multiprovider, this will be referred to as `multiprovider.MultiProvider`, which is not ideal. We could also implement the multiprovider in the openfeature package, but that will "pollute" the namespace and documentation in that package with identifiers that many users won't need. Another option is to name the package `multi`, so that you could have `multi.Provider`. OpenTelemetry sort of takes this approach. See the noop package: https://pkg.go.dev/go.opentelemetry.io/otel/trace/noop